test/distributed/_spmd/test_transformation.py

```diff
@@ -0,0 +1,135 @@
+# Owner(s): ["oncall: distributed"]
+
+from copy import deepcopy
+from functools import wraps
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch.distributed._spmd.api import compile
+from torch.distributed._spmd.gm_transformation import GraphModuleTransformation
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    with_comms as base_with_comms,
+)
+
+
+def with_comms(func):
+    @base_with_comms
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        # make sure we set different random seeds for each rank
+        # otherwise we dont need DDP / SPMD
+        # (we would have the same parameters and inputs everywhere)
+        torch.manual_seed(torch.distributed.get_rank())
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class DummyModel(nn.Module):
+    def __init__(self, layers: int, dim: int):
+        super().__init__()
+        self.mod = nn.Sequential(*((nn.Linear(dim, dim), nn.ReLU()) * layers))
```

Comment:
I see, this is where shared params are coming from :)

Reply:
Yup, :( I changed it locally, have not updated the PR yet. :